src/main/java/frc/team2412/robot/subsystem/IntakeSubsystem.java

```diff
@@ -1,142 +1,214 @@
-        intakeMotorState = IntakeMotorState.STOPPED;
+
+        this.leftColorSensor = leftColorSensor;
+        this.rightColorSensor = rightColorSensor;
+        this.centerColorSensor = rightColorSensor;
+
+        colorMatcher.setConfidenceThreshold(0.9);
+
+        // Looking for opposite alliance ball color to extake
+        if (teamColor == Alliance.Blue) {
+            colorMatcher.addColorMatch(RED_CARGO_COLOR);
+        } else if (teamColor == Alliance.Red) {
+            colorMatcher.addColorMatch(BLUE_CARGO_COLOR);
+        }
+
+        intakeSolenoidState = RETRACT;
+        intakeMotorState = STOPPED;
 
         intakeRetract();
         intakeStop();
 
         setName("IntakeSubsystem");
     }
 
-    // Methods.
+    // Methods
 
+    /**
+     * Spins motors inwards and updates motor state
+     */
     public void intakeIn() {
-        if (intakeSolenoidState == IntakeSolenoidState.EXTENDED) {
-            motorOuterAxle.set(IntakeConstants.INTAKE_IN_SPEED);
-            motorInnerAxle.set(IntakeConstants.INTAKE_IN_SPEED);
+        if (intakeSolenoidState == EXTEND) {
+            motorOuterAxle.set(INTAKE_IN_SPEED);
+            motorInnerAxle.set(INTAKE_IN_SPEED);
 
-            intakeMotorState = IntakeMotorState.IN;
+            intakeMotorState = IN;
         }
     }
 
+    /**
+     * Spins motors outwards and updates motor state
+     */
     public void intakeOut() {
-        if (intakeSolenoidState == IntakeSolenoidState.EXTENDED) {
-            motorOuterAxle.set(IntakeConstants.INTAKE_OUT_SPEED);
-            motorInnerAxle.set(IntakeConstants.INTAKE_OUT_SPEED);
+        if (intakeSolenoidState == IntakeSolenoidState.EXTEND) {
+            motorOuterAxle.set(INTAKE_OUT_SPEED);
+            motorInnerAxle.set(INTAKE_OUT_SPEED);
 
-            intakeMotorState = IntakeMotorState.OUT;
+            intakeMotorState = OUT;
         }
     }
 
+    /**
+     * Stops motors and updates motor state
+     */
     public void intakeStop() {
         motorOuterAxle.set(0);
         motorInnerAxle.set(0);
 
-        intakeMotorState = IntakeMotorState.STOPPED;
+        intakeMotorState = STOPPED;
     }
 
+    /**
+     * Extends solenoid and updates solenoid state
+     */
     public void intakeExtend() {
-        intakeSolenoidState = IntakeSolenoidState.EXTENDED;
+        intakeSolenoidState = EXTEND;
 
-        solenoid.set(IntakeState.EXTEND.value);
+        solenoid.set(EXTEND.value);
     }
 
+    /**
+     * Retracts solenoid and updates solenoid state
+     */
     public void intakeRetract() {
-        intakeSolenoidState = IntakeSolenoidState.RETRACTED;
+        intakeSolenoidState = RETRACT;
+
+        solenoid.set(RETRACT.value);
+    }
 
-        solenoid.set(IntakeState.RETRACT.value);
+    /**
+     * Returns true if color sensor detects an enemy cargo
+     */
+    public boolean individualHasOpposingColorCargo(MultiplexedColorSensor colorSensor) {
```

Comment:
Should be `private`, as code outside this class shouldn't have access to the MultiplexedColorSensor objects.

Reply:
If there's a need to know left/center/right, then consider adding those functions.  Though I'd recommend against it, as I'm not sure why they'd be needed (and it's best to not speculatively add code and increase support/maintenance cost).